### ISSUE TITLE
1. Scanning frequencies can be scheduled (e.g. once, daily, weekly, monthly) or run ad-hoc

### ISSUE THREAD

**Issue by madelondohmen** [minvws/nl-kat-coordination#3065]
### Milestone item
1 scanning frequencies.

### Detailed description
_Add detailed description of the new feature you'd like to propose. If this feature is related to a problem, what problem does it solve?_

### Scope
* _Item1_
* _Item2_

### Out of scope 
* _Item1_
* _Item2_

## Related tickets
- [x] #204
- [x] #2779 
- [x] #2974

**Comment by jpbruinsslot** [comment]
Can the scope and requirements for this ticket be fleshed out?

**Comment by underdarknl** [comment]
Users should be able to create (for a single organisation) a new report -recipe.
This report recipe should then lead the scheduler to periodically (as specified in the recipe), execute a Report-job in a report-job executor.
This Report job for now means asking Rocky to create the Report as specified by the recipe.
The user should then be able to see and interact with reports generated by this flow, and also create/update/delete the recipes.

**Comment by jpbruinsslot** [comment]
@underdarknl does this mean:

A: a user defines an interval schedule on a boefje detail page for that particular boefje to run?
B: a user defines an interval schedule for a boefje _and_ ooi combination?

**Comment by underdarknl** [comment]
It means Case A.
The boefje itself will carry a default interval in its boefje.json, which can be overwritten by the user on a 'per organisation' basis.

Scanning individual assets with the same boefje on different interval's is not part of the MVP.

**Comment by jpbruinsslot** [comment]
Implementation suggestion:

rocky pov:
- update interval of plugin
  - `PATCH` associated schedule for plugin in scheduler

scheduler pov:
 - cold start
   - when the schedulers start it will gather all enabled plugins
   - for every plugin with a defined interval create a schedule
 - enabled/disabled plugins
   - whenever a plugin gets enabled with a defined interval create a schedule
   - whenever a plugin gets disabled with a defined interval remove (disabled) its associated schedule

**Comment by underdarknl** [comment]
related: https://github.com/minvws/nl-kat-coordination/issues/1317

**Comment by underdarknl** [comment]
The boefje.json is stored *per* install, and as such should not be changed when dealing with a new interval.
The specific boefje settings are stored per organisation (eg, is it enabled, what other settings are needed), and should be updated.

**Comment by jpbruinsslot** [comment]
- Updated the implementation suggestion above
- Created poc pull request here for the scheduler https://github.com/minvws/nl-kat-coordination/pull/3406

blocked by unimplemented `boefje.json`

**Comment by underdarknl** [comment]
I think this has now been implemented?

**Comment by jpbruinsslot** [comment]
> I think this has now been implemented?

yes
